java/src/main/java/com/nedap/retail/example/rest/ApiCaller.java

```diff
@@ -125,6 +125,8 @@ private String doHttpRequest(final String url, final String requestMethod, final
             }
         } else if (responseCode == 401) {
             throw new UnauthorizedException("Unauthorized");
+        } else if (responseCode == 404) {
```

Comment:
Don't we have some Http constant for `404`? E.g. `HttpURLConnection.HTTP_NOT_FOUND`
Also for other status codes